### ISSUE TITLE
extra_hosts resolve container name to ip

### ISSUE THREAD

**Issue by gimler** [docker/compose#4100]
It would be nice if i can use a containername instate of a ip in `extra_hosts` and docker resolve the ip for me.

example:
I have a project with multiple container now i would start a phantomjs container with a domain alias

```
#!/usr/bin/env sh

# grab container ip
IP=`docker-compose exec webserver ip add show eth1 | grep global | awk -F '[/ ]+' '{print $3}'`

# start phantomjs with the alias
docker run -d -v ${PWD}:/phantomjs --net=schoolm_default --add-host="www.example.local:${IP}" --name=phantomjs fgrehm/phantomjs2 --config /phantomjs/phantomjs-config.json
```

**Comment by shin-** [comment]
This looks like the sort of use-case you'd want to use custom [networks](https://docs.docker.com/compose/networking/) for. That way you would be able to access other containers using hostnames and wouldn't have to bother with `extra_hosts` at all.

**Comment by gimler** [comment]
Thanks for the hint i have forgot that network aliases exists ;)

**Comment by namedgraph** [comment]
@shin- what if want to map `localhost` to a service IP?

I'm using
```
    extra_hosts:
      - localhost:${NGINX_IP}
```
but this is PITA because the IP can change.

I tried putting `localhost` as `alias` on the nginx service, but within other services in the network `localhost` resolves to `127.0.0.1` regardless.
